catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -479,6 +480,39 @@ public void init() {
           return util.getJson(response);
         });
 
+    put(
+        "/sharing/:id",
+        (req, res) -> {
+          String id = req.params(":id");
+
+          Map<String, Object> queryTemplate =
+              JsonFactory.create().parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard = util.getMetacard(id);
+
+          if (isShareableMetacard(metacard)) {
+            metacard.setAttribute(
+                new AttributeImpl(
+                    SecurityAttributes.ACCESS_INDIVIDUALS,
+                    getSecurityAttributes(queryTemplate, SecurityAttributes.ACCESS_INDIVIDUALS)));
+
+            metacard.setAttribute(
+                new AttributeImpl(
+                    SecurityAttributes.ACCESS_GROUPS,
+                    getSecurityAttributes(queryTemplate, SecurityAttributes.ACCESS_GROUPS)));
+
+            updateMetacard(id, metacard);
+
+            // TODO: Change this response message to something that has meaning
+            return util.getJson("Updated");
```

Comment:
Can't return metacard here since `workspaces` and `templates` run through different transformers.. What should I return as the response for all generic shareable metacard types here?

Reply:
I would return some sort of result object that describes the outcome of the operation:
```
ImmutableMap.of("message", "Metacard was updated successfully", "success", true);
```
or
```
ImmutableMap.of("message", "Could not update sharing permissions for metacard", "success", false);
```